Mod2/Assessments/Week2Assessment.md

```diff
@@ -0,0 +1,62 @@
+# Week 2 Assessment
+
+## Questions
+1. What does TDD stand for?
+
+1. What are three benefits of using TDD?
+
+1. Imagine you are in an interview.  The interviewer asks: How do you use TDD?
+
+1. For the class below, outline the tests you would need.  Try to use as much C# syntax as possible. The first test has been provided for you.
+```c#
+public class Dog
+    {
+        public string Name { get; }
+        public string Breed { get; }
+        public bool IsHungry { get; private set; }
+
+        public Dog(string name, string breed)
+        {
+            Name = name;
+            Breed = breed;
+            IsHungry = true;
+        }
+
+        public void Eat()
+        {
+            IsHungry = false;
+        }
+
+        public void Sleep()
+        {
+            IsHungry = true;
+        }
+
+        public string Speak()
+        {
+            return "Bark Bark!";
+        }
+    }
+```
+```c#
+// Add your tests here
+
+[Fact]
+public void DogHasNameAttribute()
+{
+    Dog dog = new Dog("Nile", "Golden Retriever");
+
+    Assert.Equal("Nile", dog.Name);
+}
+```
+
+1. What is a merge conflict, and when might you encounter one?
+
+1. You and a partner are working on a project together.  Your partner is working on aa-branch; you are working on bb-branch.  In as much detail as possible, describe how you both would get your work combined onto the main branch.
+
+1. Why is it important to never merge your own branch (when working on a team)?
```

Comment:
I'd prefer for us to not talk in such absolutes about never merging your own branch. All of the work flows I've used on the job involve someone else on the team approving when they think the PR looks good enough to merge. Then the owner of the PR making any last changes and merging themself. Or they put it back up for more reviews if more changes are needed.

Maybe we change the question to something like "Why is it good practice to have someone else approve and/or merge your PR?"

Reply:
Yep, totally agree